src/Microsoft.VisualStudio.Composition/ComposedPart.cs

```diff
@@ -40,7 +42,18 @@ public ComposedPart(ComposablePartDefinition definition, ImmutableDictionary<Imp
         /// <summary>
         /// Gets a map of this part's imports, and the exports which satisfy them.
         /// </summary>
-        public ImmutableDictionary<ImportDefinitionBinding, IReadOnlyList<ExportDefinitionBinding>> SatisfyingExports { get; private set; }
+        public IReadOnlyDictionary<ImportDefinitionBinding, IReadOnlyList<ExportDefinitionBinding>> SatisfyingExports
+        {
+            get
+            {
+                return this.satisfyingExports;
+            }
+
+            private set
+            {
+                this.satisfyingExports = ImmutableDictionary.CreateRange(value);
+            }
```

Comment:
But you don't even need a setter, since we now have a field that you set directly. So this whole property can be written as:

```cs
public IReadOnlyDictionary<ImportDefinitionBinding, IReadOnlyList<ExportDefinitionBinding>> SatisfyingExports => this.satisfyingExports;
```